tools/mk_vendor

```diff
@@ -0,0 +1,854 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+NAME="mk_vendor"
+VERSION="0.1"
+
+USAGE="Usage:
+    ${NAME} make <configuration> [vendor-options]
+    ${NAME} locate <configuration> [vendor-options]
+    ${NAME} -h | --help
+    ${NAME} --version
+
+build and interact with vendor libraries
+
+Commands:
+    Get help for commands with ${NAME} COMMAND --help
+
+    make    build the vendor library
+    locate  find a vendor library
+
+Common Options:
+    -h, --help  display help and exit
+    --version   display version and exit
+
+Examples:
+    build the default version of OpenSSL:
+      \$ ${NAME} make openssl
+
+    build OpenSSL 1.1.1u:
+      \$ ${NAME} make openssl:openssl111u
+
+    locate the prefix directory of OpenSSL 3.1.2 with asan
+      \$ ${NAME} locate openssl:openssl312 --options=asan
+"
+
+MAKE_USAGE="Usage:
+    ${NAME} make <configuration>[:<preset>] [vendor-options]
+    ${NAME} make -h | --help
+
+fetch, patch, build and install a vendor library
+
+Vendor Options:
+    -n,--name=STRING       override the configuration name
+    -f,--fetch=STRING      override the fetch method used to retrieve the sources
+    -p,--patch=STRING,...  override the patch sets applied to the sources
+    -b,--build=STRING,...  override the build scripts configuration
+    --options=STRING,...   override the build options
+
+Other Options:
+    -h, --help  display help and exit
+    --dry-run   stop after reading the configuration file
+
+Examples:
+    build the default version of OpenSSL:
+      \$ ${NAME} make openssl
+
+    build OpenSSL 1.1.1u with patched CVE and ASAN:
+      \$ ${NAME} make openssl:openssl111u --patch=all_CVE --options=asan
+
+    build OpenSSL 1.1.1j and give it a different name:
+      \$ ${NAME} make openssl:openssl111j --name=openssl-default
+"
+
+LOCATE_USAGE="Usage:
+    ${NAME} locate <configuration>[:<preset>] [vendor-options]
+    ${NAME} locate -h | --help
+
+Display the path to the install prefix that would be created by passing the same
+vendor configuration to '${NAME} make'.
+
+Vendor Options:
+    -n,--name=STRING       override the configuration name
+    -f,--fetch=STRING      override the fetch method used to retrieve the sources
+    -p,--patch=STRING,...  override the patch sets applied to the sources
+    -b,--build=STRING,...  override the build scripts configuration
+    --options=STRING,...   override the build options
```

Comment:
Can you give examples of which kind of values are possible here?

Reply:
These options match the structure of the configuration in `openssl.toml`. Each allows you to override the default value set in the presets.

For example: `mk_vendor make openssl:openssl111k --fetch=openssl111j` will override the sources fetch to use the `fetch.openssl111j` value defined in the configuration file instead of the one defined in the selected preset (`fetch.openssl111k`).

In truth, most of the time you won't need to use these options but use the defined presets instead. So I don't want to write too much about them in the help text. Maybe a better approach would be to guide the user through an interactive CLI to help her define a custom preset. This would allow us to provide the list of fetch/build/... values already present in the specific configuration file that she wants to extend.